### DISCUSSION TITLE
How to insert iconDiv at the end of each paragraph

### DISCUSSION THREAD

**Discussion by mahmedmushtaq** [facebook/lexical#2682] [Q&A] [answered]
Hi Lexical team, Actually I want to insert some icons at the end of each paragraph but I am lost on how can I do this. I tried a different ways but it is not working. An example is attached of what I want to achieve in a lexical editor
<img width="549" alt="image" src="https://user-images.githubusercontent.com/61344963/180437055-dfe867db-3859-441b-bd5e-fa551beb3a3e.png">

**Comment by thegreatercurve** [comment, marked as answer]
I've not tested this, but you could do something like the below to get the bounding box of the second-most parent element for each node in the tree, and then use that to position to absolutely position a floating menu.
```js
const anchorNode = $getSelection().anchorNode;

const secondLevelParentNode = $findMatchingParent(anchorNode, node => $getDepth(node) === 1)

const matchingElementForSecondLevelParentNode = editor.getElementByKey(secondLevelParentNode.getKey())

const {x, y} = matchingElementForSecondLevelParentNode.getBoundingClientRect()
```
You could then use `registerUpdateListener` to recalculate the positions, any time there is an update to editor state.
